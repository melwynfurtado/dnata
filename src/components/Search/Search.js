import React from 'react'
import PropTypes from 'prop-types'
import Checkbox from './Checkbox'

const Search = ({ 
  query, 
  handleOnQueryChange, 
  handleOnSearch, 
  handleOnStarsChange, 
  handleOnMinCostChange, 
  handleOnUserRatingChange, 
  minCost 
}) => {
  return (
    <div className="jumbotron jumbotron-fluid">
      <div className="container">
        <h2>Hotels</h2>
        <p className="lead">Search best hotel deals available.</p>
        <form onSubmit={handleOnSearch}>
          <div className="form-row">
            <div className="form-group col-md-6">
              <label className="sr-only" htmlFor="hotelName">Hotel Name</label>
              <input 
                type="text" 
                className="form-control mb-2 mr-sm-2" 
                id="hotelName" 
                placeholder="Search by hotel name" 
                value={query} 
                onChange={handleOnQueryChange} 
              />
            </div>
            <div className="form-group col-md-6">
              <button type="submit" className="btn btn-primary mb-2">Search</button>
            </div>
          </div>
          <a className="advance-search" data-toggle="collapse" href="#collapseExample" aria-expanded="false" aria-controls="collapseExample">
            Advanced Search
          </a>       

          <div className="form-row collapse" id="collapseExample">              
            <div className="form-group col-md-3 mr-2">
              <label className="text-muted">Stars:</label>
              <Checkbox id="5star" value="5" handleChange={handleOnStarsChange}>5 stars</Checkbox>
              <Checkbox id="4star" value="4" handleChange={handleOnStarsChange}>4 stars</Checkbox>
              <Checkbox id="3star" value="3" handleChange={handleOnStarsChange}>3 stars</Checkbox>
              <Checkbox id="2star" value="2" handleChange={handleOnStarsChange}>2 stars</Checkbox>
              <Checkbox id="1star" value="1" handleChange={handleOnStarsChange}>1 stars</Checkbox>
              <Checkbox id="unrated" value="0" handleChange={handleOnStarsChange}>Unrated</Checkbox>
            </div>

            <div className="form-group col-md-3 mr-2">
              <label className="text-muted">User Rating:</label>
              <Checkbox id="spectacular" value="10" handleChange={handleOnUserRatingChange}>
                Spectacular <small><strong>(10)</strong></small>
              </Checkbox>
              <Checkbox id="exceptional" value="9" handleChange={handleOnUserRatingChange}>
                Exceptional <small><strong>(9+)</strong></small>
              </Checkbox>
              <Checkbox id="excellent" value="8" handleChange={handleOnUserRatingChange}>
                Excellent <small><strong>(8+)</strong></small>
              </Checkbox>
              <Checkbox id="2star" value="7" handleChange={handleOnUserRatingChange}>
                Good <small><strong>(7+)</strong></small>
              </Checkbox>
              <Checkbox id="1star" value="6" handleChange={handleOnUserRatingChange}>
                Above Average <small><strong>(6+)</strong></small>
              </Checkbox>
              <Checkbox id="unrated" value="0" handleChange={handleOnUserRatingChange}>
                Unrated
              </Checkbox>                                        
            </div>

            <div className="form-group col-md-3 mr-2">
              <label className="text-muted" htmlFor="minCost">Maximum cost (per night):</label>
              <div className="input-group">
                <div className="input-group-prepend">
                  <span className="input-group-text">Â£</span>
                </div>
                <input 
                  type="number" 
                  id="minCost" 
                  name="minCost" 
                  className="form-control" 
                  placeholder="0" 
                  aria-label="Amount (to the nearest pound)" 
                  value={minCost} 
                  onChange={handleOnMinCostChange} 
                />
              </div>             
            </div>                 
          </div>         
        </form>        
      </div>
    </div>    
  )
}

Search.propTypes = {
  query: PropTypes.string,
  handleOnQueryChange: PropTypes.func,
  handleOnMinCostChange: PropTypes.func,
  handleOnSearch: PropTypes.func,
  handleOnStarsChange: PropTypes.func,
  handleOnUserRatingChange: PropTypes.func,  
  minCost: PropTypes.string,
}

export default Search