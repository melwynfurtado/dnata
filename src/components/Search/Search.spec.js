import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Search from './Search'

describe('Search Component', () => {
  let handleOnQueryChange
  let Component

  beforeEach(() => {
    handleOnQueryChange = jest.fn()
    Component = <Search handleOnQueryChange={handleOnQueryChange} />
  })

  it('renders correctly', () => {
    const tree = renderer
                  .create(Component)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls handleOnQueryChange on query change', () => {
    const wrapper = shallow(Component)
    wrapper.find('#hotelName').simulate('change')
    expect(handleOnQueryChange).toHaveBeenCalled()
  })
})