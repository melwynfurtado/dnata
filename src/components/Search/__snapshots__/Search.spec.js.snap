// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component renders correctly 1`] = `
<div
  className="jumbotron jumbotron-fluid"
>
  <div
    className="container"
  >
    <h2>
      Hotels
    </h2>
    <p
      className="lead"
    >
      Search best hotel deals available.
    </p>
    <form>
      <div
        className="form-row"
      >
        <div
          className="form-group col-md-6"
        >
          <label
            className="sr-only"
            htmlFor="hotelName"
          >
            Hotel Name
          </label>
          <input
            className="form-control mb-2 mr-sm-2"
            id="hotelName"
            onChange={[MockFunction]}
            placeholder="Search by hotel name"
            type="text"
          />
        </div>
        <div
          className="form-group col-md-6"
        >
          <button
            className="btn btn-primary mb-2"
            type="submit"
          >
            Search
          </button>
        </div>
      </div>
      <a
        aria-controls="collapseExample"
        aria-expanded="false"
        className="advance-search"
        data-toggle="collapse"
        href="#collapseExample"
      >
        Advanced Search
      </a>
      <div
        className="form-row collapse"
        id="collapseExample"
      >
        <div
          className="form-group col-md-3 mr-2"
        >
          <label
            className="text-muted"
          >
            Stars:
          </label>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="5star"
              type="checkbox"
              value="5"
            />
            <label
              className="form-check-label"
              htmlFor="5star"
            >
              5 stars
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="4star"
              type="checkbox"
              value="4"
            />
            <label
              className="form-check-label"
              htmlFor="4star"
            >
              4 stars
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="3star"
              type="checkbox"
              value="3"
            />
            <label
              className="form-check-label"
              htmlFor="3star"
            >
              3 stars
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="2star"
              type="checkbox"
              value="2"
            />
            <label
              className="form-check-label"
              htmlFor="2star"
            >
              2 stars
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="1star"
              type="checkbox"
              value="1"
            />
            <label
              className="form-check-label"
              htmlFor="1star"
            >
              1 stars
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="unrated"
              type="checkbox"
              value="0"
            />
            <label
              className="form-check-label"
              htmlFor="unrated"
            >
              Unrated
            </label>
          </div>
        </div>
        <div
          className="form-group col-md-3 mr-2"
        >
          <label
            className="text-muted"
          >
            User Rating:
          </label>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="spectacular"
              type="checkbox"
              value="10"
            />
            <label
              className="form-check-label"
              htmlFor="spectacular"
            >
              Spectacular 
              <small>
                <strong>
                  (10)
                </strong>
              </small>
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="exceptional"
              type="checkbox"
              value="9"
            />
            <label
              className="form-check-label"
              htmlFor="exceptional"
            >
              Exceptional 
              <small>
                <strong>
                  (9+)
                </strong>
              </small>
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="excellent"
              type="checkbox"
              value="8"
            />
            <label
              className="form-check-label"
              htmlFor="excellent"
            >
              Excellent 
              <small>
                <strong>
                  (8+)
                </strong>
              </small>
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="2star"
              type="checkbox"
              value="7"
            />
            <label
              className="form-check-label"
              htmlFor="2star"
            >
              Good 
              <small>
                <strong>
                  (7+)
                </strong>
              </small>
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="1star"
              type="checkbox"
              value="6"
            />
            <label
              className="form-check-label"
              htmlFor="1star"
            >
              Above Average 
              <small>
                <strong>
                  (6+)
                </strong>
              </small>
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="1star"
              type="checkbox"
              value="5"
            />
            <label
              className="form-check-label"
              htmlFor="1star"
            >
              Average 
              <small>
                <strong>
                  (5+)
                </strong>
              </small>
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="1star"
              type="checkbox"
              value="4"
            />
            <label
              className="form-check-label"
              htmlFor="1star"
            >
              Below Average 
              <small>
                <strong>
                  (4+)
                </strong>
              </small>
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="1star"
              type="checkbox"
              value="3"
            />
            <label
              className="form-check-label"
              htmlFor="1star"
            >
              Unsatisfactory 
              <small>
                <strong>
                  (3+)
                </strong>
              </small>
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="1star"
              type="checkbox"
              value="2"
            />
            <label
              className="form-check-label"
              htmlFor="1star"
            >
              Poor 
              <small>
                <strong>
                  (2+)
                </strong>
              </small>
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="1star"
              type="checkbox"
              value="1"
            />
            <label
              className="form-check-label"
              htmlFor="1star"
            >
              Very Poor 
              <small>
                <strong>
                  (1+)
                </strong>
              </small>
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              className="form-check-input"
              id="unrated"
              type="checkbox"
              value="0"
            />
            <label
              className="form-check-label"
              htmlFor="unrated"
            >
              Unrated
            </label>
          </div>
        </div>
        <div
          className="form-group col-md-3 mr-2"
        >
          <label
            className="text-muted"
            htmlFor="minCost"
          >
            Maximum cost (per night):
          </label>
          <div
            className="input-group"
          >
            <div
              className="input-group-prepend"
            >
              <span
                className="input-group-text"
              >
                Â£
              </span>
            </div>
            <input
              aria-label="Amount (to the nearest pound)"
              className="form-control"
              id="minCost"
              name="minCost"
              placeholder="0"
              type="number"
            />
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
